### ISSUE TITLE
Java 9 support

### ISSUE THREAD

**Issue by johnpoth** [jibx/jibx#9]
when building with Java 9 build 156 I get the error:
```
[ERROR] Failed to execute goal org.jibx:maven-jibx-plugin:1.3.1:test-bind (default) on project camel-jibx: Superclass java.lang.Object of class org.jibx.runtime.Utility not found -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.jibx:maven-jibx-plugin:1.3.1:test-bind (default) on project camel-jibx: Superclass java.lang.Object of class org.jibx.runtime.Utility not found
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:212)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:307)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:193)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:106)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:863)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:288)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:199)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:543)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Caused by: org.apache.maven.plugin.MojoExecutionException: Superclass java.lang.Object of class org.jibx.runtime.Utility not found
	at org.jibx.maven.AbstractBaseBindingMojo.compile(AbstractBaseBindingMojo.java:166)
	at org.jibx.maven.AbstractBaseBindingMojo.execute(AbstractBaseBindingMojo.java:133)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:207)
	... 20 more
Caused by: java.lang.IllegalArgumentException: Superclass java.lang.Object of class org.jibx.runtime.Utility not found
	at org.jibx.binding.def.StringConversion.<init>(StringConversion.java:163)
	at org.jibx.binding.def.PrimitiveStringConversion.<init>(PrimitiveStringConversion.java:143)
	at org.jibx.binding.def.BindingDefinition.<clinit>(BindingDefinition.java:92)
	at org.jibx.binding.Compile.compile(Compile.java:210)
	at org.jibx.maven.AbstractBaseBindingMojo.compile(AbstractBaseBindingMojo.java:163)
	... 23 more
```

To reproduce, build [camel-jibx](https://github.com/apache/camel/tree/master/components/camel-jibx). I haven't found a workaround yet.

**Comment by manishoswalsynerzip** [comment]
Was anyone able to resolve this issue . I am also getting similar error while upgrading from Java 8 to Java 9 
Please see the below error logs : 

[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 20.979s
[INFO] Finished at: Wed Jan 17 13:55:45 IST 2018
[INFO] Final Memory: 17M/58M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.jibx:maven-jibx-plugin:1.3.1:bind (default) on project eps-common: Superclass java.lang.Object of class org.jibx.runtime.Utility not found -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.jibx:maven-jibx-plugin:1.3.1:bind (default) on project eps-common: Superclass java.lang.Object of class org.jibx.runtime.Utility not found
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:217)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:84)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:59)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.singleThreadedBuild(LifecycleStarter.java:183)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:161)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:320)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:156)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:537)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:196)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:141)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:290)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:230)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:409)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:352)
Caused by: org.apache.maven.plugin.MojoExecutionException: Superclass java.lang.Object of class org.jibx.runtime.Utility not found
	at org.jibx.maven.AbstractBaseBindingMojo.compile(AbstractBaseBindingMojo.java:166)
	at org.jibx.maven.AbstractBaseBindingMojo.execute(AbstractBaseBindingMojo.java:133)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:101)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:209)
	... 19 more
Caused by: java.lang.IllegalArgumentException: Superclass java.lang.Object of class org.jibx.runtime.Utility not found
	at org.jibx.binding.def.StringConversion.<init>(StringConversion.java:163)
	at org.jibx.binding.def.PrimitiveStringConversion.<init>(PrimitiveStringConversion.java:143)
	at org.jibx.binding.def.BindingDefinition.<clinit>(BindingDefinition.java:92)
	at org.jibx.binding.Compile.compile(Compile.java:210)
	at org.jibx.maven.AbstractBaseBindingMojo.compile(AbstractBaseBindingMojo.java:163)
	... 22 more
[ERROR] 
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException

**Comment by SanjayGautam** [comment]
I do get this issue showing while validating the pom.xm , it shows up in eclipse as error , but the target source compile is still JDK 8 and hence compiling without any issue.  

Superclass java.lang.Object of class org.jibx.runtime.Utility not found (org.jibx:jibx-maven-plugin:1.3.1:bind:default:process-classes)

org.apache.maven.plugin.MojoExecutionException: Superclass java.lang.Object of class org.jibx.runtime.Utility not found
	at org.jibx.maven.AbstractBaseBindingMojo.compile(AbstractBaseBindingMojo.java:166)

**Comment by abergmann85** [comment]
Has anyone of you found a solution to this Problem ?

**Comment by hlavki** [comment]
It works with latest version of bcel 6.3

```xml
                <plugin>
                    <groupId>org.jibx</groupId>
                    <artifactId>maven-jibx-plugin</artifactId>
                    <version>1.3.1</version>
                    <executions>
                        <execution>
                            <goals>
                                <goal>bind</goal>
                            </goals>
                        </execution>
                    </executions>
                    <dependencies>
                        <dependency>
                            <groupId>org.apache.bcel</groupId>
                            <artifactId>bcel</artifactId>
                            <version>6.3</version>
                        </dependency>
                    </dependencies>
                </plugin>
```

**Comment by ElyesKhama** [comment]
Has anyone been able to run Jibx with Java9 ? 
I've got the same issue even with updating all versions to latest.

**Comment by doncorley** [comment]
This issue has been resolved with the release of JiBX 1.3.3.
Closed.
